changelog.d/12098.bugfix

```diff
@@ -0,0 +1 @@
+Fix a bug introduced in Synapse 1.51.0rc1 where incoming federation transactions containing at least one EDU would be dropped if debug logging was enabled for `synapse.8631_debug`. Synapse 1.53.0rc1 included a partial fix this ([#11890](https://github.com/matrix-org/synapse/pull/11890)), but the fix was incorrect: EDUs containing device list or signing key updates would still be dropped.
```

Comment:
Not sure it's worth mentioning #11890 since it didn't end up fixing it?

Reply:
I just want to clarify that I've not accidentally copied and pasted a bugfix line from 1.53's release notes.

Reply:
Fair enough, though I'd tend to think looking at the PR is enough to tell that